client/component/people.js

```diff
@@ -54,49 +53,91 @@ function renderLayers(type, persona, profilePhotoLayers) {
+export function MyProfilePhoto({type='large', photo=null, faint=false, border=false}) {
     const personaPreview = usePersonaPreview();
     if (personaPreview?.photoUrl) {
         return <FaceImage photoUrl={photo ?? personaPreview.photoUrl} type={type} faint={faint} 
-            border={border} check={check} />
+            border={border} />
     } else if (personaPreview?.hue && personaPreview?.name) {
         return <LetterFace name={personaPreview.name} hue={personaPreview.hue} type={type} />
     } else {
         return <AnonymousFace faint={faint} type={type} border={border} />
     }
 }
 
-export function FaceImage({ face, photoUrl = null, type = 'small', faint = false, check = false, border = false}) {
-    
+export function UnverifiedBadge({size = 14}) {
+    const s = UnverifiedBadgeStyles;
+    return (
+        <View style={[s.container, { width: size, height: size, borderRadius: size / 2 }]}>
+            <Text style={[s.text, { fontSize: size === 24 ? 18 : 12 }]}>?</Text>
+        </View>
+    );
+}
+
+const UnverifiedBadgeStyles = StyleSheet.create({
+    container: {
+        backgroundColor: colorGreyHover,
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    text: {
+        color: 'black',
+    },
+});
+
+function Badge({type, name}) {
+    const datastore = useDatastore();
+    const fbUser = datastore.getFirebaseUser();
+    const personaPreview = usePersonaPreview();
+    const verificationStatus = personaPreview?.verificationStatus ?? null;
+    const badgeSize = badgeSizeMap[type ?? 'small'];
+
+    if (fbUser?.displayName !== name) {
+        return null;
+    }
+
+    if (verificationStatus === 'verified') {
+        return <IconCircleCheck size={badgeSize} />;
+    }
+    if (verificationStatus === 'unverified') {
+        return <UnverifiedBadge size={badgeSize} />;
+    }
+    return null;
+}
+
+export function FaceImage({ face, photoUrl = null, type = 'small', faint = false, border = false, name = null }) {
     const size = sizeMap[type] ?? 32;
-    const checkPad = check ? checkPadMap[type ?? 'small'] : 0;
+    const badgePad = badgePadMap[type ?? 'small'];
 
     return <View style={{ position: 'relative', alignSelf: 'flex-start' }}>
         <Image
             style={{
                 width: size, height: size, borderRadius: size / 2,
-                opacity: faint ? 0.5 : 1, marginRight: checkPad,
+                opacity: faint ? 0.5 : 1,
                 borderWidth: border ? 2 : 0,
                 borderColor: 'white'
             }}
             source={{ uri: photoUrl ?? ('https://new-public-demo.web.app/faces/' + face) }} />
-        {check && <View style={{ position: 'absolute', right: 0, bottom: 0 }}>
-            <IconCircleCheck />
-        </View>}
+        <View style={{ position: 'absolute', right: -badgePad, bottom: 0 }}>
+            <Badge type={type} name={name} />
```

Comment:
This call isn't saying what user the badge should be shown for. The badge should be the status of the user whose face is being shown, not of the logged in user.

Reply:
fixed.